proposals/NNNN-cloud_client_library_phase_1.md

```diff
@@ -0,0 +1,139 @@
+
+
+![SDL as separate service](../assets/proposals/NNNN-cloud_client_library_phase_1/cloud_arch_1.png "SDL as separate service")
+
+##### SDL Cloud as Embedded App
+
+Lastly, the library could be deployed in an embedded app that ran on the IVI system or a different module that had access to the local network for Core to connect to the app.
+
+
+![SDL as embedded app](../assets/proposals/NNNN-cloud_client_library_phase_1/cloud_arch_2.png "SDL as embedded app")
+
+#### Code base
+
+##### Code Architecture
+
+The library would follow the SDL Android library. This would include the new manager APIs that have been created to align with the iOS platform. The idea is to keep this as similar to the android integration as possible so that developers don't feel overwhelmed if moving from one to the other and keeping as much of their integration as aligned as possible.
+
+##### Proposed project hierarchy:
+
+![Project Hierarchy](../assets/proposals/NNNN-cloud_client_library_phase_1/project_hierarchy.png "Proposed project hierarchy")
+
+
+The solution will have three defined milestones that should be implemented over a span of time.
+
+1. Refactor and extract SDL Android library to pure Java
+2. Create `sdl_java_common` library 
+3. Implement `sdl_java_common` back into SDL Android Library
+
+
+### Refactor and extract SDL Android library to pure Java
+
+The cloud/embedded java library will first be created by taking the SDL Android library and refactoring only the necessary parts to create a pure java implementation.
+
+The idea will be to create a single library that is refactored. This could be released as an Alpha/Beta version in order to get the library out soon, but allow us to prepare for potential breaking changes when creating the `sdl_java_common` library. 
+
+#### Refactor where it makes sense
+
+There are areas of the library that could use refactoring either from poor original implementation/design, incorrect terminology, etc. This would be the ideal time to take care of those issues.
+
+Some of those issues are as follows:
+
+- `SessionType.java` This is supposed to be Service type. 
+- `ByteEnumer.java` This implementation is orphaned and should be removed in favor of standard Enums
+- Any references to PCM or H264/NAV as service types. They should be `AudioStreaming` and `VideoStreaming` 
+- Remove any deprecated code
+
+
+
+### Create `sdl_java_common` library 
+
+Once a working a refactored library has been created, a new library will be extracted from that code base to become the `sdl_java_common` library. This would take into account platform specific APIs that can be used and how each of the classes can be extended in a way that works for all. 
```

Comment:
Change `Once a working a refactored library has been created` to `Once a working refactored library has been created,`